### ISSUE TITLE
when insert same character with the patternSymbol, will happen crash!

### ISSUE THREAD

**Issue by ericchuhong** [luximetr/AnyFormatKit#39]
eg: current textfield exist "8123456#"，then, if your insert other number，will crash。Console show 

Fatal error: String index is out of bounds: file Swift/StringCharacterView.swift, line 60

**Comment by luximetr** [comment]
> eg: current textfield exist "8123456#"，then, if your insert other number，will crash。Console show
> 
> Fatal error: String index is out of bounds: file Swift/StringCharacterView.swift, line 60

Will investigate it, can you please share how do you create a formatter, that crashes.

**Comment by ericchuhong** [comment]
Here is my code to create formatter.
```swift
···
/// SubClass inheritance from TextFieldInputController
class MeetingNumberInputDelegator: TextFieldInputController {
    override func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
        /// filter space in string
        let inputStr = string.replacingOccurrences(of: " ", with: "", options: .literal, range: nil)
        return super.textField(textField, shouldChangeCharactersIn: range, replacementString: inputStr)
    }
}
···
/// Define in view
var meetingNumberInputController = MeetingNumberInputDelegator()
var meetingNumberFormatter : DefaultTextInputFormatter = {
      let formatter = DefaultTextInputFormatter(textPattern: "### #### ########")
      return formatter
}()
···
/// Use
inputNoField.delegate = meetingNumberInputController
meetingNumberInputController.formatter = meetingNumberFormatter
```

**Comment by luximetr** [comment]
> eg: current textfield exist "8123456#"，then, if your insert other number，will crash。Console show
> 
> Fatal error: String index is out of bounds: file Swift/StringCharacterView.swift, line 60

I can suggest you a quick fix, during init of formatter you can pass pattern symbol parameter and set it to some other symbol. Later I will try to fix this crash.

**Comment by luximetr** [comment]
Fixed at [2.5.2](https://github.com/luximetr/AnyFormatKit/releases/tag/2.5.2).
